src/criteria_assessment/tiles.jl

```diff
@@ -64,74 +64,29 @@ function _lon_lat_to_tile(zoom, lon, lat)
     return x, y
 end
 
-"""
-    nearest(rst::Raster, tile_size::Tuple{Int, Int})::Matrix
-
-Resample a raster to a tile size using nearest neighbor interpolation.
-This approach prioritising performance over accuracy.
-
-# Arguments
-- `rst`: The input raster to be resampled.
-- `tile_size`: The desired dimensions of the tile (lat, long).
-
-# Returns
-Matrix with the resampled data.
-
-# Examples
-```julia
-large_raster = Raster(rand(UInt8, 14756, 14838); dims=(X(1:1:14756), Y(1:1:14838)))
-small_matrix = nearest(large_raster, (256, 256))
-```
-"""
-function nearest(rst::Raster, tile_size::Tuple{Int,Int})::Matrix
-    old_size = size(rst)
-
-    # Important: must flip axes!
-    # Rasters.jl stores longitude along first dimension (rows) by default.
-    x_ratio = old_size[1] / tile_size[2]
-    y_ratio = old_size[2] / tile_size[1]
-
-    resampled = zeros(eltype(rst), tile_size)
-
-    Threads.@threads for lat in 1:tile_size[1]
-        for lon in 1:tile_size[2]
-            # Use area averaging for downsampling
-            x_start = max(1, floor(Int, (lon - 1) * x_ratio) + 1)
-            x_end = min(old_size[1], ceil(Int, lon * x_ratio))
-            y_start = max(1, floor(Int, (lat - 1) * y_ratio) + 1)
-            y_end = min(old_size[2], ceil(Int, lat * y_ratio))
-
-            count_val = count(rst[x_start:x_end, y_start:y_end].data .> 0)
-            if count_val == 0
-                continue
-            end
-
-            sum_val = sum(rst[x_start:x_end, y_start:y_end].data)
-            resampled[lat, lon] = ceil(sum_val / count_val)
-        end
-    end
-
-    return resampled
-end
+# Helper functions for Web Mercator projection
+_lat_to_y(lat::Float64) = log(tan(π / 4 + lat * π / 360))
```

Comment:
these don't already exist in a geo library?

Reply:
I was interacting with GDAL via Rasters.jl and the performance hit was very significant (talking minutes).

I could potentially call out to GDAL.jl directly, but I think it would be doing extra things we don't need for our immediate use case.

Mark as future improvement?

Reply:
Would this still be needed if we switched to web-mercator for all the data in the future?